compiler/rustc_mir_dataflow/src/impls/liveness.rs

```diff
@@ -179,3 +183,115 @@ impl DefUse {
         }
     }
 }
+
+/// Like `MaybeLiveLocals`, but does not mark locals as live if they are used in a dead assignment.
+///
+/// This is basically written for dead store elimination and nothing else.
+///
+/// All of the caveats of `MaybeLiveLocals` apply.
+pub struct MaybeTransitiveLiveLocals<'a>(&'a BitSet<Local>);
+
+impl<'a> MaybeTransitiveLiveLocals<'a> {
+    /// The `always_alive` set is the set of locals to which all stores should unconditionally be
+    /// considered live.
+    ///
+    /// This should include at least all locals that are ever borrowed.
+    pub fn new(always_alive: &'a BitSet<Local>) -> Self {
+        MaybeTransitiveLiveLocals(always_alive)
+    }
+}
+
+impl<'a, 'tcx> AnalysisDomain<'tcx> for MaybeTransitiveLiveLocals<'a> {
+    type Domain = BitSet<Local>;
```

Comment:
There is a large 50% max-rss regression in keccack. Replacing BitSet with a ChunkedBitSet would probably help there.